### PR TITLE
deprecate lr

### PR DESCRIPTION
**Pull request by hhbyyh**
This PR is to refine the API for NNEstimator/NNClassifier.
NNEstimator and NNClassifier now supports setting learning rate via setLearningRate, yet users can also set learning rate (and learning rate decay) with OptimMehod, which can lead to confusion. Deprecate setLearningRate and setLearningRateDecay.

### CODE REVIEW

**Comment by hhbyyh** [comment]
@dding3 can you please review this? Thanks.

File: pyzoo/zoo/pipeline/nnframes/nn_classifier.py [hunk 1]
```diff
@@ -256,6 +256,7 @@ def getEndWhen(self):
     def setLearningRate(self, val):
         """
         Sets the value of :py:attr:`learningRate`.
+        .. note:: Deprecated in 4.0. Please set learning rate with optimMethod directly.
```
Comment (dding3) [inline pyzoo/zoo/pipeline/nnframes/nn_classifier.py]: Deprecated in 0.4.0?

File: pyzoo/zoo/pipeline/nnframes/nn_classifier.py [hunk 2]
```diff
@@ -269,6 +270,7 @@ def getLearningRate(self):
     def setLearningRateDecay(self, val):
         """
         Sets the value of :py:attr:`learningRateDecay`.
+        .. note:: Deprecated in 4.0. Please set learning rate decay with optimMethod directly.
```
Comment (dding3) [inline pyzoo/zoo/pipeline/nnframes/nn_classifier.py]: Deprecated in 0.4.0?

File: pyzoo/zoo/pipeline/nnframes/nn_classifier.py (hunk 1 above)
Comment (hhbyyh) [reply pyzoo/zoo/pipeline/nnframes/nn_classifier.py]: Yes, is our next release 0.4.0?

Comment (hhbyyh) [reply pyzoo/zoo/pipeline/nnframes/nn_classifier.py]: Ah I see. Thanks.

**Comment by hhbyyh** [comment]
http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/1438/ passed. Thanks for the review. Merging to master.

**Pull request merged**